src/client/sound/ogg.c

```diff
@@ -709,3 +710,62 @@ OGG_Shutdown(void)
 
 	ogg_started = false;
 }
+
+void
+OGG_LoadAsWav(char *filename, wavinfo_t *info, void **buffer)
+{
+	void * temp_buffer = NULL;
+	int size = FS_LoadFile(filename, &temp_buffer);
+	short *final_buffer = NULL;
+	stb_vorbis * ogg_file = NULL;
+	int res = 0;
+
+	if (!temp_buffer)
+	{
+		/* no such file */
+		return;
+	}
+
+	/* load vorbis file from memory */
+	ogg_file = stb_vorbis_open_memory(temp_buffer, size, &res, NULL);
+	if (!res)
+	{
+		int read_samples = 0;
+
+		/* fill in wav structure */
+		info->rate = ogg_file->sample_rate;
+		info->width = 2;
+		info->channels = ogg_file->channels;
+		info->loopstart = -1;
+		info->samples = stb_vorbis_stream_length_in_samples(ogg_file);
+		info->dataofs = 0;
+
+		/* alloc memory for uncompressed wav */
+		final_buffer = Z_Malloc(info->samples * sizeof(short));
```

Comment:
shouldn't this also take the number of channels into account?

Reply:
I have changed it in 3cc39bb48e3b42847e6e5f6024e7c460f83ce114, looks like stb_vorbis_stream_length_in_samples has returned samples multiplied by channels.